Intersect.Client/Entities/Player.cs

```diff
@@ -2560,27 +2678,77 @@ private partial class TargetInfo
+                    PacketSender.SendDirection((byte)Dir);
+                    Globals.Me.MoveDir = Direction.None;
                 }
 
                 // Hold the player in place if the requested direction is the same as the current one.
                 if (!Globals.Me.IsMoving && Dir == Globals.Me.MoveDir)
                 {
-                    Globals.Me.MoveDir = -1;
+                    Globals.Me.MoveDir = Direction.None;
+                }
+            }
+        }
+        
+        // Checks if the target is at the opposite direction of the current player's direction.
+        // The comparison also takes into account whether diagonal movement is enabled or not.
+        private static bool IsTargetAtOppositeDirection(Direction currentDir, Direction targetDir)
+        {
+            if (Options.Instance.MapOpts.EnableDiagonalMovement)
+            {
+                // If diagonal movement is enabled, check for the opposite direction for 8 directions.
+                switch (currentDir)
+                {
+                    case Direction.Up:
+                        return targetDir == Direction.Down || targetDir == Direction.DownLeft ||
+                               targetDir == Direction.DownRight;
+                    case Direction.Down:
+                        return targetDir == Direction.Up || targetDir == Direction.UpLeft ||
+                               targetDir == Direction.UpRight;
+                    case Direction.Left:
+                        return targetDir == Direction.Right || targetDir == Direction.UpRight ||
+                               targetDir == Direction.DownRight;
+                    case Direction.Right:
+                        return targetDir == Direction.Left || targetDir == Direction.UpLeft ||
+                               targetDir == Direction.DownLeft;
+                    case Direction.UpLeft:
+                        return targetDir == Direction.DownRight;
+                    case Direction.UpRight:
+                        return targetDir == Direction.DownLeft;
+                    case Direction.DownLeft:
+                        return targetDir == Direction.UpRight;
+                    case Direction.DownRight:
+                        return targetDir == Direction.UpLeft;
+                    default:
+                        return false;
                 }
             }
+
+            // If diagonal movement is disabled, check for the opposite direction for 4 directions.
+            switch (currentDir)
+            {
+                case Direction.Up:
+                    return targetDir == Direction.Down;
+                case Direction.Down:
+                    return targetDir == Direction.Up;
+                case Direction.Left:
+                    return targetDir == Direction.Right;
+                case Direction.Right:
+                    return targetDir == Direction.Left;
+                default:
+                    return false;
+            }
```

Comment:
Move this switch case first, and in the default case do if Diagonal return false, otherwise break

Then drop the redundant cases from the original switch case

Reply:
self reminder: double check this tomorrow.